community/lms/doctype/lesson/lesson.py

```diff
@@ -43,3 +43,54 @@ def get_prev(self):
         The return value would be like 1.2, 2.1 etc.
         It will be None if there is no next lesson.
         """
+
+    def get_progress(self):
+        return frappe.db.get_value("LMS Course Progress", {"lesson": self.name, "owner": frappe.session.user}, "status")
+
+    def get_slugified_class(self):
+        if self.get_progress():
+            return ("").join([ s for s in self.get_progress().lower().split() ])
+        return
+
+@frappe.whitelist()
+def save_progress(lesson):
+    if frappe.db.exists("LMS Course Progress",
+            {
+                "lesson": lesson,
+                "owner": frappe.session.user
+            }):
+        return
+
+    lesson_details = frappe.get_doc("Lesson", lesson)
+    dynamic_content = frappe.db.count("LMS Section",
+                        filters={
+                            "type": ["not in", ["example", "text"]],
+                            "parent": lesson_details.name
+                        })
+
+    status = "Complete"
+    if dynamic_content:
+        status = "Partially Complete"
```

Comment:
Shouldn't you call `all_dynamic_content_submitted` function to check if all the execises are complete?

Reply:
Here I just check if there is dynamic content in the lesson, submission is checked only through update_progress